pkg/registry/bundlegraphloader.go

```diff
@@ -0,0 +1,122 @@
+package registry
+
+import (
+	"fmt"
+
+	"github.com/blang/semver"
+)
+
+type BundleGraphLoader struct {
+}
+
+// TODO: add logger
+// TODO: add semver mode parameter -- implement skippatch graph update
+// AddBundleToGraph takes a bundle and an existing graph and updates the graph to insert the new bundle
+// into each channel it is included in
+func (g *BundleGraphLoader) AddBundleToGraph(bundle *Bundle, graph *Package, newDefaultChannel string) (*Package, error) {
+	bundleVersion, err := bundle.Version()
+	if err != nil {
+		return nil, fmt.Errorf("Unable to extract bundle version from bundle %s, can't insert in semver mode", bundle.BundleImage)
+	}
+
+	versionToAdd, err := semver.Make(bundleVersion)
+	if err != nil {
+		return nil, fmt.Errorf("Bundle version %s is not valid", bundleVersion)
+	}
+
+	newBundleKey := BundleKey{
+		CsvName:    bundle.Name,
+		Version:    versionToAdd.String(),
+		BundlePath: bundle.BundleImage,
+	}
+
+	// for each channel the new bundle is being insert into
+	for _, channel := range bundle.Channels {
```

Comment:
nit: this is getting long with lots of nested conditionals / loops, could use a bit of cleanup